astropy/io/fits/hdu/base.py

```diff
@@ -1311,55 +1285,39 @@ def verify_checksum(self, blocking='standard'):
 
         if 'CHECKSUM' in self._header:
             if 'DATASUM' in self._header:
-                datasum = self._calculate_datasum(blocking)
+                datasum = self._calculate_datasum()
             else:
                 datasum = 0
-            checksum = self._calculate_checksum(datasum, blocking)
+            checksum = self._calculate_checksum(datasum)
             if checksum == self._header['CHECKSUM']:
                 return 1
-            elif blocking == 'either':
-                # i.e. standard failed,  try nonstandard
-                return self.verify_checksum(blocking='nonstandard')
             else:
-                # Failed with all permitted blocking kinds
+                # Failed
                 return 0
         else:
             return 2
 
-    def _verify_checksum_datasum(self, blocking):
+    def _verify_checksum_datasum(self):
         """
         Verify the checksum/datasum values if the cards exist in the header.
         Simply displays warnings if either the checksum or datasum don't match.
         """
 
-        # NOTE:  private data members _checksum and _datasum are
-        # used by the utility script "fitscheck" to detect missing
-        # checksums.
-
         if 'CHECKSUM' in self._header:
             self._checksum = self._header['CHECKSUM']
-            self._checksum_comment = self._header.comments['CHECKSUM']
-            if not self.verify_checksum(blocking):
+            self._checksum_valid = self.verify_checksum()
+            if not self._checksum_valid:
```

Comment:
The code is correct, and it's perfectly valid Python, but in terms of style and clarity it bothers me a little bit that `_checksum_valid` and `_datasum_valid` are being used as booleans, even though `verify_checksum`returns integer values. It would be a little more explicit to have the line above be something like
`self._checksum_valid = self._verify_checksum() == 1`

Even better would to have named constants for the return values of `verify_checksum` and `verify_datasum`:
```
CHECKSUM_INVALID = 0
CHECKSUM_VALID = 1
CHECKSUM_MISSING = 2
```